geode-core/src/main/java/com/gemstone/gemfire/management/internal/cli/shell/Gfsh.java

```diff
@@ -279,7 +281,12 @@ protected Gfsh(boolean launchShell, String[] args, GfshConfig gfshConfig) throws
   private void initializeEnvironment() {
     env.put(ENV_SYS_USER,              System.getProperty("user.name"));
     env.put(ENV_SYS_USER_HOME,         System.getProperty("user.home"));
-    env.put(ENV_SYS_HOST_NAME,         System.getProperty("user.name"));
+    try {
+      env.put(ENV_SYS_HOST_NAME,       InetAddress.getLocalHost().getHostName());
```

Comment:
Use SocketCreator.getLocalHost() and SocketCreator.getHostName().  SocketCreator caches host names to avoid going to DNS when the host name is already known.